### PR TITLE
[sbt 1.0] server reboot

### PR DESCRIPTION
**Pull request by eed3si9n**
This implements the sbt server outlined in http://eed3si9n.com/sbt-server-reboot.

Pending https://github.com/sbt/util/pull/48

### CODE REVIEW

File: reset.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+rm -rf ~/.sbt/boot/scala-2.11.8/org.scala-sbt/sbt/1.0.0-SNAPSHOT/
```
Comment (Duhemm) [inline reset.sh:3]: I guess this was not committed on purpose?

**Comment by Duhemm** [comment]
Should we migrate that to sjson-new and sbt-datatype before merging?

File: reset.sh (hunk 1 above)
Comment (eed3si9n) [reply reset.sh:3]: That was on purpose. We can put it in `scripts` directory or something to avoid clutter.

**Comment by eed3si9n** [comment]
@Duhemm Good idea.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ object Dependencies {
 
   // sbt modules
   val ioVersion = "1.0.0-M6"
-  val utilVersion = "0.1.0-M13"
+  val utilVersion = "0.1.0-M14-SNAPSHOT"
```
Comment (dwijnand) [inline project/Dependencies.scala]: 0.1.0-M14 has been cut

**Review by dwijnand — CHANGES_REQUESTED** [review]

**Comment by eed3si9n** [comment]
Migrated to sbt-datatype

**Comment by eed3si9n** [comment]
Rebasing on top of 1.0.x to resolve conflicts.

File: main-command/src/main/datatype/server.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      "type": "record",
+      "target": "Scala",
+      "fields": [
+        {
+          "name": "type",
+          "type": "String",
+          "since": "0.0.0"
+        },
+        {
+          "name": "command_line",
```
Comment (dwijnand) [inline main-command/src/main/datatype/server.json]: This should be commandLine

File: main-command/src/main/datatype/server.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          "since": "0.0.0"
+        },
+        {
+          "name": "status",
+          "type": "String?",
+          "default": "None",
+          "since": "0.1.0"
+        },
+        {
+          "name": "command_queue",
```
Comment (dwijnand) [inline main-command/src/main/datatype/server.json]: commandQueue

**Review by dwijnand — CHANGES_REQUESTED** [review]

**Comment by eed3si9n** [comment]
There was a rebase artifact in one of the commits, so I fixed that.

File: main-command/src/main/datatype/server.json (hunk 3 above)
Comment (eed3si9n) [reply main-command/src/main/datatype/server.json]: Done

**Review by dwijnand — APPROVED** [review]

**Comment by eed3si9n** [comment]
codacy is mad, but Travis is green.

**Pull request merged**